ai-catalog/ui-component/src/main/java/com/android/ai/uicomponent/Buttons.kt

```diff
@@ -0,0 +1,201 @@
+ * limitations under the License.
+ */
+package com.android.ai.uicomponent
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccountBox
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.android.ai.theme.AISampleCatalogTheme
+
+@Composable
+fun PrimaryButton(
+    text: String,
+    modifier: Modifier = Modifier,
+    contentColor: Color = MaterialTheme.colorScheme.onPrimary,
+    containerColor: Color = MaterialTheme.colorScheme.primary,
+    icon: ImageVector? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        modifier = modifier.height(48.dp),
+        colors = ButtonDefaults.buttonColors(
+            contentColor = contentColor,
+            containerColor = containerColor,
+        ),
+        onClick = { onClick() },
+    ) {
+        if (icon != null) {
+            Image(
+                imageVector = icon,
+                contentDescription = null,
+                colorFilter = ColorFilter.tint(contentColor),
+                modifier = Modifier.size(width = 24.dp, height = 24.dp),
+            )
+            Spacer(modifier = Modifier.width(8.dp))
+        }
+        Text(
+            text = text,
+            style = MaterialTheme.typography.labelLarge,
+            modifier = Modifier.padding(end = 8.dp),
+        )
+    }
+}
+
+@Preview
+@Composable
+fun PrimaryButtonSmallPreview() {
+    AISampleCatalogTheme {
+        PrimaryButton(
+            text = "Primary button",
+            icon = Icons.Default.AccountBox,
+            onClick = {},
+        )
+    }
+}
+
+@Composable
+fun GenerateButton(
+    text: String,
+    modifier: Modifier = Modifier,
+    contentColor: Color = MaterialTheme.colorScheme.onTertiary,
+    containerColor: Color = MaterialTheme.colorScheme.tertiary,
+    enabled: Boolean = true,
+    icon: ImageVector? = ImageVector.vectorResource(id = R.drawable.draw_auto),
```

Comment:
You can just use `painterResource`, slightly cleaner. You'll have to update the name of the parameter in Image as well